### PR TITLE
Fix client.extractDomain

### PR DESCRIPTION
**Pull request by m-cat**
Expand `extractDomainForPortal` to work for input URLs in addition to domains. Add tests.

### CODE REVIEW

**Review by dghelm — APPROVED** [review]
Lgtm

File: src/registry.ts [hunk 1]
```diff
@@ -135,13 +135,11 @@ export async function getEntry(
     // TODO: Refactor this validation into a separate function.
     /* istanbul ignore next */
     if (!err.response) {
-      console.log(`Full error: ${err}`);
-      throw new Error("Error response not found");
+      throw new Error(`Error response not found. Full error: ${err}`);
```
Comment (peterjan) [inline src/registry.ts:138]: I bumped into this one, it's a bit of a weird error to bump into without context. Can we rephrase it to be more clear as to what we mean here?

**Review by peterjan — APPROVED** [review]
LGTM 👍  thanks for removing the console logs
